multiraft/raft.go

```diff
@@ -40,7 +41,20 @@ func init() {
 //
 // This file is named raft.go instead of something like logger.go because this
 // file's name is used to determine the vmodule parameter: --vmodule=raft=1
-type raftLogger struct{}
+type raftLogger struct {
+	group uint64
+}
+
+func (r *raftLogger) prependContext(format string, v []interface{}) string {
+	var s string
+	if r.group != 0 {
+		v2 := append([]interface{}{r.group}, v...)
+		s = fmt.Sprintf("group %d "+format, v2...)
```

Comment:
ugh

On Wed, Aug 12, 2015 at 11:28 AM, Tobias Schottdorf <
notifications@github.com> wrote:

> In multiraft/raft.go
> <https://github.com/cockroachdb/cockroach/pull/2038#discussion_r36873627>:
>
> > @@ -40,7 +41,20 @@ func init() {
> >  //
> >  // This file is named raft.go instead of something like logger.go because this
> >  // file's name is used to determine the vmodule parameter: --vmodule=raft=1
> > -type raftLogger struct{}
> > +type raftLogger struct {
> > +	group uint64
> > +}
> > +
> > +func (r *raftLogger) prependContext(format string, v []interface{}) string {
> > +	var s string
> > +	if r.group != 0 {
> > +		v2 := append([]interface{}{r.group}, v...)
> > +		s = fmt.Sprintf("group %d "+format, v2...)
>
> I don't think that generally works: http://play.golang.org/p/--ojU-3WPK
>
> Might work with append([]interface{}(r.group), v) though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2038/files#r36873627>.
>